src/kw_ssh.sh

```diff
@@ -109,6 +122,146 @@ function run_command_in_the_remote()
   cmd_manager "$flag" "$ssh_compose"
 }
 
+# This function transfers a local file or directory to a
+# specified remote destination.
+function ssh_send_file_to_remote()
+{
+  local user
+  local port
+  local remote
+  local remote_file
+  local remote_file_host
+  local send_path
+  local to_path
+  local flag
+  local cmd
+  local send_option
+  local ssh_compose
+  local filename
+  local filename_path
+
+  flag=${flag:-'SILENT'}
+
+  user="${remote_parameters['REMOTE_USER']}"
+  port="${remote_parameters['REMOTE_PORT']}"
+  remote="${remote_parameters['REMOTE_IP']}"
+  remote_file="${remote_parameters['REMOTE_FILE']}"
+  remote_file_host="${remote_parameters['REMOTE_FILE_HOST']}"
+  send_path=${options_values['SEND']}
+  to_path=${options_values['TO']:-'.'}
+  flag=${options_values['VERBOSE']}
+  ssh_compose=$(handle_ssh)
+  filename=$(basename "$send_path")
+  filename_path=$(printf "%s/%s" "$to_path" "$filename" | tr --squeeze-repeats '/')
+
+  cmd_manager "$flag" "${ssh_compose} [[ -f '${to_path}' ]]"
+
+  if [[ "$?" -eq 0 ]]; then
+    complain "The '${to_path}' file already exists on the remote machine."
+    if [[ $(ask_yN 'Do you wish to overwrite it?') =~ '0' ]]; then
+      exit 22 # EINVAL
+    fi
+  fi
+
+  cmd_manager "$flag" "${ssh_compose} [[ -f '${to_path}/${filename}' ]]"
+
+  if [[ "$?" -eq 0 ]]; then
+    complain "The '${filename_path}' file already exists on the remote machine."
+    if [[ $(ask_yN 'Do you wish to overwrite it?') =~ '0' ]]; then
+      exit 22 # EINVAL
+    fi
+  fi
+
+  if [[ ! -e "$send_path" ]]; then
+    complain "The file or directory '${send_path}' does not exist."
+    exit 22 # EINVAL
+  fi
+
+  if [[ -f "$remote_file" ]]; then
+    cmd="rsync -e 'ssh -F ${remote_file}' -avz ${send_path} ${remote_file_host}:${to_path} > /dev/null 2>&1"
+  else
+    cmd="rsync -e 'ssh -p ${port}' -avz ${send_path} ${user}@${remote}:${to_path} > /dev/null 2>&1"
```

Comment:
The `-avz ${send_path} ${user}@${remote}:${to_path} > /dev/null 2>&1` is equal in the `if` and `else`. Maybe you can rewrite this code to something like this:
```
cmd="rsync -e 'ssh -F ${remote_file}"
...
if [[ ! -f "$remote_file" ]]; then
  cmd="rsync -e 'ssh -p ${port}'"
fi
cmd+=" -avz ${send_path} ${user}@${remote}:${to_path} > /dev/null 2>&1"
```

Also, why do we need `2>&1`?

Reply:
We need to use `2>&1` to redirect the output of the rsync command and prevent data such as the byte baud rate of the file transfer from being displayed.